src/jlapi.c

```diff
@@ -1073,6 +1088,10 @@ JL_DLLEXPORT int jl_repl_entrypoint(int argc, char *argv[])
     }
 
     julia_init(jl_options.image_file_specified ? JL_IMAGE_CWD : JL_IMAGE_JULIA_HOME);
+    jl_atomic_store_release(&jl_runtime_is_initialized, 1);
```

Comment:
Should this be set by `julia_init` instead?

It seems like embedding applications might call that directly